### PR TITLE
Add externaltrafficpolicy service compatibility

### PR DESCRIPTION
**Pull request by Ken-UC**
Fixes https://github.com/mariadb-operator/mariadb-operator/issues/210

Add service External_Traffic_Policy allows source ip preservation when service type = loadbalancer

### CODE REVIEW

**Comment by mmontes11** [comment]
Hey there @Ken-UC ! Thanks for the contribution 🦭  

The CI is failing, could you please take a look 🙏🏻

File: api/v1alpha1/mariadb_types.go [hunk 1]
```diff
@@ -70,6 +70,7 @@ func (p *PodDisruptionBudget) Validate() error {
 type ServiceTemplate struct {
 	Type        corev1.ServiceType `json:"type,omitempty"`
 	Annotations map[string]string  `json:"annotations,omitempty"`
+	ExternalTrafficPolicy [string]string `json:"externaltrafficpolicy,omitempty"`
```
Comment (mmontes11) [inline api/v1alpha1/mariadb_types.go:73]: This map notation doesn't seem to be accepted in our GolangCI Lint rules, could you change it to `map[string]string`?

Also, the fields should be in camel case, could you update this also?: `externalTrafficPolicy`

**Comment by Ken-UC** [comment]
also add loadBalancerSourceRanges and loadBalancerIp to match with our needs

File: api/v1alpha1/mariadb_types.go [hunk 2]
```diff
@@ -70,6 +70,9 @@ func (p *PodDisruptionBudget) Validate() error {
 type ServiceTemplate struct {
 	Type        corev1.ServiceType `json:"type,omitempty"`
 	Annotations map[string]string  `json:"annotations,omitempty"`
+	externalTrafficPolicy string `json:"externalTrafficPolicy,omitempty"`
+	loadBalancerSourceRanges []string `json:"loadBalancerSourceRanges,omitempty"`
+	loadBalancerIp string `json:"loadBalancerIp,omitempty"`
```
Comment (mmontes11) [inline api/v1alpha1/mariadb_types.go:75]: This fields need to start with capital letter to be exported. Could you update them please

**Review by mmontes11 — COMMENTED** [review]
Left a comment about field naming

**Review by mmontes11 — COMMENTED** [review]
Could you take a look at the failing CI steps?

**Comment by Ken-UC** [comment]
> Could you take a look at the failing CI steps?

Tried, but not fluent in golang :(

**Comment by mmontes11** [comment]
Closed in favor of:
- https://github.com/mariadb-operator/mariadb-operator/pull/227

I've ported your changes there and added you as a co-author. Thanks for your contribution! 🙏🏻

**Comment by Ken-UC** [comment]
many thanks for your help and for maintain this operator!

**Pull request closed without merging**